elements_ephys/readers/spikeglx.py

```diff
@@ -23,40 +23,76 @@ def __init__(self, root_dir):
         name & associated meta - no interpretation of g0_t0.imec, etc is
         performed at this layer.
         '''
+        self._apmeta, self._apdata = None, None
+        self._lfmeta, self._lfdata = None, None
 
-        self.root_dir = root_dir
+        self.root_dir = pathlib.Path(root_dir)
 
         meta_filepath = next(pathlib.Path(root_dir).glob('*.ap.meta'))
-        self.npx_meta = NeuropixelsMeta(meta_filepath)
-
         self.root_name = meta_filepath.name.replace('.ap.meta', '')
-        self._apdata = None
-        self._lfdata, self._lfmeta = None, None
+
+    @property
+    def apmeta(self):
+        if self._apmeta is None:
+            self._apmeta = SpikeGLXMeta(self.root_dir / (self.root_name + '.ap.meta'))
+        return self._apmeta
 
     @property
     def apdata(self):
```

Comment:
again, i would like to avoid `data` as name. Perhaps `ap_spikes` would be more descriptive.